tests/src/test/java/org/wso2/micro/gateway/tests/common/CLIExecutor.java

```diff
@@ -114,6 +114,43 @@ public void generateFromDefinition(String label, String project, String endpoint
         }
     }
 
+    public void generatePassingFlag(String label, String project, String additionalFlag) throws Exception {
+        org.wso2.apimgt.gateway.cli.cmd.Main main = new org.wso2.apimgt.gateway.cli.cmd.Main();
+
+        String baseDir = (System.getProperty(Constants.SYSTEM_PROP_BASE_DIR, ".")) + File.separator + "target";
+        Path path = Files.createTempDirectory(new File(baseDir).toPath(), "userProject", new FileAttribute[0]);
+        log.info("CLI Project Home: " + path.toString());
+
+        System.setProperty(GatewayCliConstants.CLI_HOME, this.cliHome);
+        log.info("CLI Home: " + this.cliHome);
+
+        String config = new File(
+                getClass().getClassLoader().getResource("confs" + File.separator + "default-cli-test-config.toml")
+                        .getPath()).getAbsolutePath();
+        System.setProperty("user.dir", path.toString());
+        String[] args = { "setup", project, "--label", label, "--username", "admin", "--password",
+                "admin", "--server-url", "http://localhost:9443", "--truststore",
```

Comment:
Shouldn't be the "localhost" configurable?

Reply:
This URL (http://localhost:9443) is the URL of the MockHttpServer used for test cases.